src/hal/lib/hal_comp.c

```diff
@@ -372,36 +372,39 @@ const char *hal_comp_name(int comp_id)
 
 // use only for owner_ids of pins, params or functs
 // may return NULL if buggy using code
+// This was a bottleneck with rcomps validating pins
+// Move checks so that instances get checked first
+
 hal_comp_t *halpr_find_owning_comp(const int owner_id)
 {
-    hal_comp_t *comp = halpr_find_comp_by_id(owner_id);
-    if (comp != NULL)
-	return comp;  // legacy case: the owner_id refers to a comp
+hal_comp_t *comp;
```

Comment:
It is a perfectly valid method of differentiating local vars with global scope within a function, from those declared within sub-scopes.
Makes it easy to check where a var came from.